provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/util/VulnerabilityUtil.java

```diff
@@ -158,6 +151,24 @@ public static LinkableItem createVulnerabilityAttributeItem(String severityValue
         return Optional.empty();
     }
 
+    private static Optional<LinkableItem> createShortTermGuidance(Supplier<ComponentVersionUpgradeGuidanceShortTermView> getShortTermOption, String remediationLabel) {
+        ComponentVersionUpgradeGuidanceShortTermView shortTermGuidance = getShortTermOption.get();
+        if (null != shortTermGuidance) {
+            String versionText = createShortTermGuidanceVersionText(shortTermGuidance);
+            return Optional.of(new LinkableItem(remediationLabel, versionText, shortTermGuidance.getVersion()));
+        }
+        return Optional.empty();
+    }
+
+    private static Optional<LinkableItem> createLongTermGuidance(Supplier<ComponentVersionUpgradeGuidanceLongTermView> getLongTermOption, String remediationLabel) {
+        ComponentVersionUpgradeGuidanceLongTermView longTermGuidance = getLongTermOption.get();
+        if (null != longTermGuidance) {
+            String versionText = createLongTermGuidanceVersionText(longTermGuidance);
+            return Optional.of(new LinkableItem(remediationLabel, versionText, longTermGuidance.getVersion()));
+        }
+        return Optional.empty();
+    }
```

Comment:
There's a lot of duplication going on in this class. Since you're working with two different types that don't share a parent, then we will probably want to create a wrapper class around ComponentVersionUpgradeGuidanceLongTermView and ComponentVersionUpgradeGuidanceLongTermView. The class would contain all the relevant shared fields with a constructor for each type where you map the share fields appropriately. Then we can have a lot of this common code in a single place.

Reply:
What do you think of the  updated method with the new models?